omegaconf/_utils.py

```diff
@@ -605,9 +605,7 @@ def is_int(st: str) -> bool:
 
 
 def is_primitive_list(obj: Any) -> bool:
-    from .base import Container
-
-    return not isinstance(obj, Container) and isinstance(obj, (list, tuple))
+    return type(obj) in (list, tuple)
```

Comment:
Won't that break code like:

```
from omegaconf import OmegaConf, ListConfig
class CustomList(list):
    pass
cfg = ListConfig(CustomList((1,2,3)))
print(cfg)
```

Reply:
Yes, you're right. I'll update the PR.

Reply:
So it turns out that the behavior on `2.1_branch` is a bit idiosyncratic: in some cases `CustomList`/`Shape` are coerced to `ListConfig` and in other cases they either rejected or wrapped in an `AnyNode` (depending on whether `allow_objects` is set).
Anyway, the new `TestCreationWithCustomClass` captures and enforces that behavior.